cosmovisor/cmd/cosmovisor/main.go

```diff
@@ -21,10 +21,12 @@ func Run(args []string) error {
 		return err
 	}
 
-	doUpgrade, err := cosmovisor.LaunchProcess(cfg, args, os.Stdout, os.Stderr)
-	// if RestartAfterUpgrade, we launch after a successful upgrade (only condition LaunchProcess returns nil)
-	for cfg.RestartAfterUpgrade && err == nil && doUpgrade {
-		doUpgrade, err = cosmovisor.LaunchProcess(cfg, args, os.Stdout, os.Stderr)
+	var doUpgrade bool
+	for {
+		if doUpgrade, err = cosmovisor.LaunchProcess(cfg, args, os.Stdout, os.Stderr); !cfg.RestartAfterUpgrade || err != nil || !doUpgrade {
+			break
+		}
```

Comment:
this change is not needed.

Reply:
When we upgrade multiple times, cosmovisor can handle this? Seems it only support single update.

Reply:
I don't think we need this, there will be single upgrade at some height.

Reply:
I mean, the node operator need to restart cosmovisor after every upgrade.

## Problem Senario
let's say, there were 5 upgrades.
v1, v2, v3, v4, v5 at h1, h2, h3, h4

When you sync from genesis, you have to restart after every second upgrade. 
```
sync1: 0 - h1 with v1

cosmovisor automatically upgrade v1 to v2
sync2: h1 - h2 with v2

cosmovisor just turned off after scanning and downloading
node operator turn on cosmovisor manually
sync3: h2 - h3 with v3

cosmovisor automatically upgrade v3 to v4
sync4: h3 - h4 with v4

cosmovisor just turned off after scanning and downloading
node operator turn on cosmovisor manually
sync5: h4 ~
```
Does it what you want?

Reply:
You can set `DAEMON_RESTART_AFTER_UPGRADE=true` and it will restart after every upgrade

Reply:
Ah I see, thanks for clarification. I though the for as if